test/unit/vl-pill.test.html

```diff
@@ -14,84 +14,139 @@
+    test('checkable pill bevat de juiste class en een input element van het type checkbox met de juiste class', () => {
+      const pill = fixture('vl-pill-fixture');
+      const shadowRoot = pill.shadowRoot;
+
+      should.not.exist(shadowRoot.querySelector(".vl-pill--checkable"));
+
+      pill.setAttribute('checkable', '');
+      const checkbox = pill._checkbox;
+      assert.isTrue(checkbox.getAttribute('type') === 'checkbox');
+      assert.isTrue(checkbox.classList.contains('vl-pill--checkable__checkbox'));
+      should.exist(shadowRoot.querySelector(".vl-pill--checkable"));
+
+      pill.removeAttribute('checkable');
+      should.not.exist(shadowRoot.querySelector("input"));
+      should.not.exist(shadowRoot.querySelector(".vl-pill--checkable"));
     });
 
-    test('checkable bevat een span element', () => {
-      const checkablePill = fixture('vl-label-pill-fixture');
-      should.exist(checkablePill.querySelector('span'));
+    test('checkable pill bevat een span element', () => {
+      const pill = fixture('vl-pill-fixture');
+      const shadowRoot = pill.shadowRoot;
+
+      pill.setAttribute('checkable', '');
+      should.exist(shadowRoot.querySelector('span'));
     });
+
+    test('checkable pill kan via "checked" uit- en aangevinkt worden', () => {
+      const pill = fixture('vl-pill-fixture');
+
+      pill.setAttribute('checkable', '');
+      pill.checked.should.be.false;
+
+      pill.checked = true;
+      pill.checked.should.be.true;
+
+      pill.checked = false;
+      pill.checked.should.be.false;
+    });
+
+    test('checkable pill kan via een klik op de pill uit- en aangevinkt worden', () => {
+      const pill = fixture('vl-pill-fixture');
+
+      pill.setAttribute('checkable', '');
+      pill.checked.should.be.false;
+
+      pill._checkbox.click();
+      pill.checked.should.be.true;
+
+      pill._checkbox.click();
+      pill.checked.should.be.false;
+    });
+
+    test('checkable pill stuurt event uit wanneer hij aan- en uitgevinkt wordt via "checked"', (done) => {
+      testCheckEvent(done, pill => pill.checked = !pill.checked);
+    });
+
+    test('checkable pill stuurt event uit wanneer hij aan- en uitgevinkt wordt via een klik op de pill', (done) => {
+      testCheckEvent(done, pill => pill._checkbox.click());
+    });
+
+    function testCheckEvent(done, toggleCheckFun) {
+      const pill = fixture('vl-pill-fixture');
+      let expectedCheckedValue = true;
+
+      pill.setAttribute('checkable', '');
+      pill.addEventListener(VlPill.EVENTS.check, function(event) {
+        event.composed.should.be.false;
+        event.bubbles.should.be.false;
+        event.detail.checked.should.equal(expectedCheckedValue);
+
+        if (expectedCheckedValue) {
+          expectedCheckedValue = false;
+          setTimeout(() => toggleCheckFun(pill));
+        } else {
+          done();
+        }
+      });
+
+      toggleCheckFun(pill);
+    }
```

Comment:
Het was even schakelen de manier waarop de test hier geschreven werd, maar omdat ik de laatste reviewer ben en niemand anders hier een opmerking op gegeven heeft is het voor mij prima.

Reply:
moet ik aan denken dan, als we ooit eens met het team een activiteit moeten doen, dat ik eerst iets voorstel dan

Reply:
Ik hou mijn hart al vast...